lib/personalized_search/product_preferences_selection.dart

```diff
@@ -5,11 +5,11 @@
 /// E.g. `SharedPreferences` and provider.
 /// Some simple app may even use simple fix solution.
 class ProductPreferencesSelection {
-  ProductPreferencesSelection(
-    this._setImportance,
-    this._getImportance,
-    this._notify,
-  );
+  ProductPreferencesSelection({
+    required this.setImportance,
+    required this.getImportance,
+    required this.notify,
```

Comment:
I think the notify callback could be optional.

Reply:
Good point! `setImportance` as well could be optional, as in my latest commit.